internal/bminventory/inventory.go

```diff
@@ -1671,6 +1681,23 @@ func (b *bareMetalInventory) updateDhcpNetworkParams(updates map[string]interfac
 	return nil
 }
 
+func (b *bareMetalInventory) updateMachineCidrOrVips(userManagedNetworking, vipDhcpAllocation bool, cluster *common.Cluster, updates map[string]interface{},
+	params installer.UpdateClusterParams, machineCidr *string, log logrus.FieldLogger) error {
+	var err error
+	updateMachineCIDRCondition := func() bool {
+		return !userManagedNetworking && vipDhcpAllocation || userManagedNetworking && len(cluster.Hosts) == 1 && network.IsIpv6OnlyHost(cluster.Hosts[0], log)
+	}
+	updateVIPsCondition := func() bool {
```

Comment:
Why is this a function?
why not just put the expression in the 'if' statement?

Reply:
Also, how come we don't need to send the CIDR for IPv4?
In that case how will etcd know on which interface to listen?

Reply:
The manifest adds settings for IPv6 only so it will have no effect on IPv4

Reply:
Added for simplicity, but it can be changed

Reply:
The CIDR protocol is ksome kind of a flag to etcd. This is the problem I tried to solve.
In the machine CIDR is not specified etcd assumes it is IPv4.  So there is a failure.

Reply:
but according to @hexfusion this is not just a matter of being a flag for IPv6. it the way that ETCD knows to which network it should listen. So this will also be needed in ipv4, no?

Reply:
IPv4 is the default - so it is not needed